src/ape_ethereum/transactions.py

```diff
@@ -50,6 +51,9 @@ class AccessList(BaseModel):
 
 
 class BaseTransaction(TransactionAPI):
+    block_id: Optional[BlockID] = None
+    state_override: Optional[Dict] = None
```

Comment:
Is this needed in the model?

Reply:
If we want to add validation to the object instead of using a dict, adding it the model is a good way to do this. Else, I don't really know where to put the types.